fiftyone/core/collections.py

```diff
@@ -1322,6 +1331,12 @@ def _do_get_dynamic_field_schema(
                 if is_list_field and not _path.endswith("[]"):
                     _path += "[]"
 
+                    # Cache the manual unwind in case we need it recursively
+                    # Insert rather than append so that nested paths are found
+                    # first when using this cache
+                    _clean_path = _path.replace("[]", "")
+                    unwind_cache.insert(0, (_clean_path, _path))
+
```

Comment:
I do not entirely understand what's going on here so just making sure it's intentional that this logic only applies when `_path.endswith("[]")` is `False`

Reply:
Yeah this is correct as written. The only way that `[]` gets into the path is inside this `if`, which is the time when we need to cache an undeclared list field that we've found.